### ISSUE TITLE
Basecamp list of available upgrades should exclude competitors to upgrades in progress

### ISSUE THREAD

**Issue by PatrikLundell** [CleverRaven/Cataclysm-DDA#45333]
# Describe the bug
When upgrading an expansion the list of available upgrades lists ones that "compete" with the one in progress. If I e.g. order the building of the first section of the central kitchen using mi-go resin, the list correctly identifies that construction as being in progress (marked with "Recover Ally"). However, the competing recipes that will be excluded upon completion of the task remain in the list, and it's even possible to order a second companion to start on the competing project (debug spawned NPC to do that). When the first project is done it's created as normal, and when the second one is done it replaces the previous one (since they are using the same tiles).

# Steps To Reproduce

Create a base camp and start an expansion (Canteen in my case). Send a companion off to build the first building using some material (I used debug spawned mi-go resin). After sending the companion off, bring up the expansion menu and see that the "same" construction for other material choices are available for selection (assuming you spawn the building materials).
It can be noted that the same thing happens with fireplaces vs stoves in the Canteen, another case where you have competing options.

# Expected behavior

Recipes that will be excluded by constructions in progress should be excluded from the list.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 2004
- Game Version: Please install `git` to generate VERSION [64-bit] (Latest experimental source zip file downloaded and compiled a few hours before this report was written, but present in earlier versions as well. I believe it was present in 0.E2 stable as well).
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]

# Additional context

Strangely enough, the the main section of the base camp seems to be handled correctly, i.e. when building beds rather than straw beds the straw beds are excluded during the construction. Likewise, starting the construction of one section of the building excluded the construction of competing ones.

Looking at the code (basecamp.cpp, basecamp::available_upgrades), it looks like there would be a need for an initial tally of the things excluded by constructions in progress and add those to the ones that have been collected for the expansion to get rid of the competing projects. Adding the excludes when projects are started instead has the problem that this would have to be reverted if the projects are cancelled.

I've looked at the JSON files for the base camp and expansion recipes, and can't see anything there that would cause the main section to behave differently from the expansions. However, since the behavior seems to be different, there should be something somewhere that causes the difference.

**Comment by PatrikLundell** [comment]
Not sure if the "suggestion" tag is the correct one, as starting competing projects in parallel is possible (and certainly possible during normal play by mistake), which leads to buggy results, so it's not just an issue of getting a correct display of upgrades.
However, the buggy results are currently not really bad (I can imaging cases where you could end up with rooms without doors or windows, for instance, but no such thing is present in the current base camp recipes, at least not for the modular one. Currently it's mostly a case of wasted resources by constructing the "same" thing twice).

**Comment by RenechCDDA** [comment]
Duplicate of #39070
